crates/net/downloaders/src/bodies/bodies.rs

```diff
@@ -253,83 +253,29 @@ where
-
-        // Dispatch requests for missing bodies
-        if let Some(latest_queued) = self.latest_queued_block_number {
-            if range.start <= latest_queued {
-                // Retrieve the range for checking whether the block has been downloaded.
-                // Since `queued_bodies` is a sorted contiguous collection of block responses,
-                // construct the range instead of iterating the array each time looking for the
-                // block number.
-                let queued_bodies_range = match (queued_bodies.first(), queued_bodies.last()) {
-                    (Some(first), Some(last)) => first.block_number()..last.block_number() + 1,
-                    _ => Range::default(),
-                };
-
-                let mut requests = Vec::<RangeInclusive<BlockNumber>>::default();
-                for num in range.start..=latest_queued {
-                    // Check if block has been downloaded or is currently in progress
-                    if queued_bodies_range.contains(&num) ||
-                        self.buffered_responses.iter().any(|r| r.block_range().contains(&num)) ||
-                        self.in_progress_queue.contains_block(num)
-                    {
-                        continue
-                    }
-
-                    match requests.last().map(|range| *range.end()) {
-                        // Extend the last range if contiguous
-                        Some(range_end) if range_end + 1 == num => {
-                            let range = requests.pop().unwrap();
-                            requests.push(*range.start()..=num);
-                        }
-                        // Push the new request range
-                        Some(_) | None => requests.push(num..=num),
-                    };
-                }
-
-                for range in requests {
-                    let headers = self
-                        .query_headers(
-                            *range.start()..*range.end() + 1,
-                            range.clone().count() as u64,
-                        )?
-                        .ok_or(DownloadError::MissingHeader { block_number: *range.start() })?;
-
-                    self.metrics.out_of_order_requests.increment(1);
-
-                    // Dispatch contiguous request.
-                    self.in_progress_queue.push_new_request(
-                        Arc::clone(&self.client),
-                        Arc::clone(&self.consensus),
-                        headers,
-                        Priority::High,
-                    );
-                }
+        // Check if the range is valid.
+        let is_valid = !range.is_empty();
+        if is_valid {
+            // Check if the provided range is the subset of the existing range.
+            let is_current_range_subset =
+                self.download_range.contains(&range.start) && range.end == self.download_range.end;
```

Comment:
does this matter?
```suggestion
                self.download_range.contains(&range.start) && range.end <= self.download_range.end;
```

Reply:
should be an unreachable case since the stage shrinks the range going forward.
e.g. 
1st iteration: range=0..100
2nd iteration: range=10..100
3rd iteration: range=20..100
etc...